### PR TITLE
Add advisors 2014-04-01

### PR DESCRIPTION
**Pull request by jaredmoo**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by jaredmoo** [comment]
@tomapaunovic please review. Also @nathannfan FYI

**Comment by jaredmoo** [comment]
The linter and model validation errors are known false positives, won't fix.

**Comment by sameralameer** [comment]
Captain Jared,

the error is about the missing operationsAPI, is it a false positive or was the decision in place to not implement it for sql?

**Comment by balajikris** [comment]
I guess its a false error, because its implemented in a different file? Jared can confirm though.

**Comment by salameer** [comment]
Oh I seeeee, @vishrutshah and @veronicagg is that a planned fix on the linter? I remember we chatted about this once.

**Comment by jaredmoo** [comment]
Yes it's in another file, https://github.com/Azure/azure-rest-api-specs/blob/master/arm-sql/2014-04-01/swagger/operations.json .

File: arm-sql/2014-04-01/swagger/advisors.json [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "name": "advisorName",
+            "in": "path",
+            "description": "The name of the Database Advisor.",
+            "required": true,
+            "type": "string"
+          },
```
Comment (balajikris) [inline arm-sql/2014-04-01/swagger/advisors.json]: extract definition to "/parameters/XXX" and reuse in other places below?

File: arm-sql/2014-04-01/swagger/advisors.json [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved details of specified database advisor.",
+            "schema": {
+              "$ref": "#/definitions/Advisor"
+            }
```
Comment (balajikris) [inline arm-sql/2014-04-01/swagger/advisors.json]: no default error response?

**Comment by balajikris** [comment]
Model validation failure is also a known false positive for this swagger. LGTM, pending Jared's response to above comments.

**Comment by veronicagg** [comment]
@sameralameer regarding your question on operationsAPI violation being reported, the mechanism is already implemented and requires calling the linter with config file and openapi-type=arm, once the repo is switched over to config files, we should start reporting this appropriately.

File: arm-sql/2014-04-01/swagger/advisors.json (hunk 1 above)
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/advisors.json]: Done

File: arm-sql/2014-04-01/swagger/advisors.json (hunk 2 above)
Comment (jaredmoo) [inline arm-sql/2014-04-01/swagger/advisors.json]: Sorry not sure what you are looking for here, can you clarify? In general we haven't been declaring the "default" response.

Comment (balajikris) [inline arm-sql/2014-04-01/swagger/advisors.json]: I see. I was wondering if you'd be specifying an error response within "default" with an appropriate code and message. Similar to what you have [here](https://github.com/Azure/azure-rest-api-specs/pull/1377/files#diff-749babe73be448583c15dfd0691c2e06R50)

**Review by balajikris — APPROVED** [review]

**Comment by jaredmoo** [comment]
Thanks!

**Pull request merged**